main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -48,6 +48,13 @@ import scala.jdk.CollectionConverters.*
   */
 object Weeder2 {
 
+  /**
+    * The maximum nesting of expressions allowed.
+    *
+    * This is here to avoid stack overflow during compiler phases.
+    */
+  private val ExpressionNestingLimit = 48
```

Comment:
Stdlib breaks for `ExpressionNestingLimit = 50`

Reply:
It seems several examples and tests go higher.

Reply:
Sorry, what i meant was, that if I have a binary expression like `1 + 1 + 1 + ..`, then this limit has to be <50 to save us from stack overflow.

The tests have a high nesting but they are using `Let` which don't generally use `map` so I think that let binders are cheaper in terms of stack size.

The bound here either has to let some stack overflows go through or limit some programs unfairly (with our nested lets/statements)

Reply:
So what's the proposal?

Reply:
Also, it seems (seen from the outside) crazy that the bound would be 50. We have computers with gigabytes of memory. How can this even be the case???

Reply:
yeah I don't know why its so low - ConstraintGen was the one to fail